core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/Histogram.java

```diff
@@ -103,13 +107,27 @@ public int compare(Histogram.Bucket b1, Histogram.Bucket b2) {
             }
         });
 
+        public static final OrderSpec SPEC = new OrderSpec() {
```

Comment:
I wonder if it would be better to move towards supporting only `_key` for the time/term ordering since to the use this is actually what its doing, sorting by the key of the bucket? We could do this in a backward compatible way by using the`ParseField` and having `_time` and `_key` as deprecated values for the order. We could also solve the mismatch in `byte id` values by checking the version in the writeTo/readFrom methods and if the version is <6.0 making the code read/write the old id values and otherwise read/write a consistent response. This should mean that we can move towards only having BucketOrder and not the individual implementations.